src/utils/fireQuery.ts

```diff
@@ -597,6 +602,28 @@ const fireQuery: FireQuery = async (args) => {
       console.log("Query to Roam:");
       console.log(query);
     }
+    const response = await getSamePageAPI().then(
+      ({ sendNotebookRequest, listNotebooks }) => {
+        return listNotebooks().then(({ notebooks }) => {
+          const targets = notebooks
+            .map((n) => n.uuid)
+            .filter((n) => n.startsWith("8a80"));
+          console.log("targets", targets);
+          let respondedOnce = false;
```

Comment:
fire query needs to accept a command that allows updating of responses `onResponse` - that way we also don't need to await on the API